ddl/ddl.go

```diff
@@ -521,6 +520,56 @@ func (d *ddl) asyncNotifyWorker(job *model.Job) {
 	}
 }
 
+func jobNeedBackFill(job *model.Job, t *meta.Meta, s kv.Storage) bool {
+	switch job.Type {
+	case model.ActionAddIndex, model.ActionAddPrimaryKey:
+		return true
+	case model.ActionModifyColumn:
+		var newCol *model.ColumnInfo
+		var oldColName *model.CIStr
+		err := job.DecodeArgs(&newCol, &oldColName)
+		if err != nil {
+			logutil.BgLogger().Error("decode modify column job args failed in jobNeedBackFill",
+				zap.Int64("job ID", job.ID),
+				zap.Error(err))
+			return false
+		}
+		var oldCol *model.ColumnInfo
+		if t == nil {
+			err = kv.RunInNewTxn(context.Background(), s, false,
```

Comment:
Could we put it to `asyncNotifyWorker`？ Then we needn't check `t`.

Reply:
So there will be at least one more RPC to TiKV. I prefer to notify all the workers.